src/append/rolling_file/append.rs

```diff
@@ -21,6 +23,47 @@ use crate::non_blocking::NonBlocking;
 use crate::Diagnostic;
 use crate::Layout;
 
+#[derive(Debug)]
+pub struct BlockingRollingFile {
+    layout: Box<dyn Layout>,
+    writer: Mutex<RollingFileWriter>,
+}
+
+impl BlockingRollingFile {
+    /// Creates a new [`BlockingRollingFile`] appender.
+    ///
+    /// This appender by default uses [`TextLayout`] to format log records.
+    pub fn new(writer: RollingFileWriter) -> Self {
+        Self {
+            layout: Box::new(TextLayout::default().no_color()),
+            writer: Mutex::new(writer),
+        }
+    }
+
+    /// Sets the layout used to format log records.
+    pub fn with_layout(mut self, layout: impl Into<Box<dyn Layout>>) -> Self {
+        self.layout = layout.into();
+        self
+    }
+}
+
+impl Append for BlockingRollingFile {
+    fn append(&self, record: &Record, diagnostics: &[Diagnostic]) -> anyhow::Result<()> {
+        let mut writer = self.writer.lock().unwrap_or_else(|e| e.into_inner());
+        let mut bytes = self.layout.format(record, diagnostics)?;
+        bytes.push(b'\n');
+        Write::write_all(&mut *writer, bytes.as_slice())?;
+        Ok(())
+    }
+
+    fn flush(&self) {
```

Comment:
@andylokandy maybe flush should return `anyhow::Result<()>` and we handle error all at one level up:

```rust
    fn log(&self, record: &Record) {
        for dispatch in &self.dispatches {
            if let Err(err) = dispatch.log(record) {
                handle_error(record, err);
            }
        }
    }

    fn flush(&self) {
        for dispatch in &self.dispatches {
            if let Err(err) = dispatch.flush() {
                handle_flush_error(err);
            }
        }
    }
```

also maybe add a ErrorSink trait for custom error listener.

Reply:
I think we can keep the behavior in sync with non-blocking its variant. is the non blocking one return result on flush?

Reply:
The non-blocking one do flush in the worker thread where the error is handled as `eprintln!("failed to write log: {err}");`.